File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.js
/**
 * @param {string} word
 * @return {number}
 */
var numberOfSpecialChars = function(word) {
     const chars = new Set(word);
    let count = 0;

    for (let char of chars) {
        if (char >= 'a' && char <= 'z' && chars.has(char.toUpperCase())) {
            count++;
        }
    }
    return count;
};

File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.cpp
class Solution {
public:
    int numberOfSpecialChars(string word) {
        set<char> chars(word.begin(), word.end());
        int count = 0;
        for (char ch : chars) {
            if (ch >= 'a' && ch <= 'z' && chars.count(toupper(ch))) {
                count++;
                
            }
        }
        return count;
    }
};